### ISSUE TITLE
Expected interplay between zero_grad and virtual_step 

### ISSUE THREAD

**Issue by ftramer** [pytorch/opacus#143]
It would be nice to have a piece of the opacus documentation that clearly describes the expected interplay between calls to `zero_grad`, `step` and `virtual_step`.

In my original implementation of virtual steps, I was using the following training loop (with opacus 0.9.0):

```python
for i, (data, target) in enumerate(train_loader):

    optimizer.zero_grad()
        
    # ... compute loss ...
    loss.backward()

    if ((i + 1) % n_accumulation_steps == 0) or ((i + 1) == len(train_loader)):
        optimizer.step()
    else:
        optimizer.virtual_step()
```

When switching to opacus 0.11.0, this flow somehow broke and I'm not getting the same results anymore.
The [CIFAR-10 example](https://github.com/pytorch/opacus/blob/master/examples/cifar10.py#L69) seems to indicate that `zero_grad()` should now be called only after every call to `step()`.

Am I correct in assuming that `virtual_step()` now silently erases all per-sample gradient information so that an explicit call to `zero_grad()` is no longer needed?

**Comment by Darktex** [comment]
Hi Florian! We updated the docstring for [virtual_step](https://opacus.ai/api/privacy_engine.html#opacus.privacy_engine.PrivacyEngine.virtual_step) (see also #96 on this).

Short answer - yes. Just call `zero_grad` after step :)
